### ISSUE TITLE
`RetryHTTPException` Should Inherit from `SimpleHTTPException `

### ISSUE THREAD

**Issue by lmahoney1** [ibmresilient/resilient-python-api#32]
Previously when using the `self.rest_client()` object within an IBM SOAR function it would raise a `SimpleHTTPException` exception when it encountered a web request error. Must have been a recent change but I'm seeing that certain response codes are getting re-tried and if all retries fail a `RetryHTTPException` is raised.

I think this is great, however all of my IBM SOAR functions I have created are setup to catch `SimpleHTTPException`s and not `RetryHTTPException`.

If `RetryHTTPException` inherited from `SimpleHTTPException` it would still get caught by

```python
except SimpleHTTPException as e:
   log.info(<information about request / response>)
```

Not 100% positive that `SimpleHTTPException` would be the best 'root' / generic web request to inherit from, but it would work best for me.

I can change my except blocks to be 

```python
except (SimpleHTTPException, RetryHTTPException) as e:
   ...
```

but I would rather not have to change that in all of our functions. I think it makes logical sense for `RetryHTTPException` to inherit from a 'generic' http exception, just not 100% positive if `SimpleHTTPException` fits that bill or not.

Additionally, importing these two exceptions is different:

```python
from resilient import SimpleHTTPException
from resilient.co3base import RetryHTTPException
```

Noticed that `SimpleHTTPException` is imported in [`/resilient/__init__.py`](https://github.com/ibmresilient/resilient-python-api/blob/84e8c6d9140ceac0bf47ce0b98e11c7953d95e61/resilient/resilient/__init__.py) but RetryHTTPException isn't. I think a consistent import would be nice - so something like `from resilient import SimpleHTTPException, RetryHTTPException` would work

`RetryHTTPException` definition: https://github.com/ibmresilient/resilient-python-api/blob/84e8c6d9140ceac0bf47ce0b98e11c7953d95e61/resilient/resilient/co3base.py#L72

**Comment by bobleckelibm** [comment]
@lmahoney1 this is a fantastic idea. I will begin investigation on this in the coming release timeframe and let you know whether or not we think this is possible. On first glance, I can't see why we wouldn't do this.

**Comment by bobleckelibm** [comment]
@lmahoney1 released today to PyPi! `v50.1.262`

**Comment by lmahoney1** [comment]
Great news. Thanks for the work on this Bo, much appreciated!
